SDK v2 migration guide.md

```diff
@@ -13,16 +13,30 @@ are outlines of the notable breaking changes as well as a mapping from version 1
    - [IoT hub service client](#iot-hub-service-client)
    - [Device Provisioning Service (DPS) device client](#dps-device-client)
    - [DPS service client](#dps-service-client)
-   - [Security provider client](#authentication-provider-client)
+   - [Security provider client](#security-provider-client)
  - [Frequently asked questions](#frequently-asked-questions)
 
 ## Why the version 1 SDK is being replaced
 
 There are a number of reasons why the Azure IoT SDK team chose to do a major version revision. Here are a few of the more important reasons:
-  - Removing or upgrading several NuGet dependencies (TODO: list).
-  - Consolidate IoT hub service clients and rename to reflect the items or operations they support.
+  - Removing or upgrading several NuGet dependencies.
+    - Upgraded
+      - Microsoft.Azure.Devices.Client
+      - Microsoft.Azure.Devices
+      - Microsoft.Azure.Devices.Authentication
+      - Microsoft.Azure.Devices.Provisioning.Client
+      - Microsoft.Azure.Devices.Provisioning.Service
+    - Removed
+      - Microsoft.Azure.Devices.Shared
+      - Microsoft.Azure.Devices.Provisioning.Transport.Amqp
+      - Microsoft.Azure.Devices.Provisioning.Transport.Http
+      - Microsoft.Azure.Devices.Provisioning.Transport.Mqtt
+      - Microsoft.Azure.Devices.Provisioning.Security.Tpm
+  - Consolidating IoT hub service clients and renaming to reflect the items or operations they support.
     - Many existing client classes (RegistryManager, ServiceClient, etc.) were confusingly named and contained methods that weren't always consistent with the client's assumed responsibilities.
   - Many existing clients had a mix of standard constructors (`new DeviceClient(...)`) and static builder methods (`DeviceClient.CreateFromConnectionString(...)`) that caused some confusion among users. The factory methods have been removed and the addition of constructors in clients enables unit testing.
+  - Exception handling. (Exception handling in the v1 SDK required multiple exception types to be caught. This logic has been consolidated to always throw only `IotHubClientException` for exceptions arising from communication attempts with IoT Hub. Parameter validation will throw `ArgumentException`/`ArgumentNullException`/`FormatException`/`InvalidOperationException`.
```

Comment:
```suggestion
  - Exception handling.
     - Exception handling in the v1 SDK required multiple exception types to be caught. This logic has been consolidated to always throw `IotHubClientException` and `IotHubServiceException` for exceptions arising from communication attempts with IoT Hub, `DeviceProvisioningClientException` and `DeviceProvisioningServiceException` for exceptions arising from communication attempts with DPS. Parameter validation will throw `ArgumentException`/`ArgumentNullException`/`FormatException`/`InvalidOperationException`.
```

Reply:
Also, currently we still have several other types of exception thrown by our clients, see [here](https://github.com/Azure/azure-iot-sdk-csharp/pull/2887#issuecomment-1281563556).  Do we want to consolidate them as well? @drwill-ms @abhipsaMisra @timtay-microsoft

Reply:
Document them here? Yes.

Reply:
> Document them here?

No, more for the question whether we want to map those exceptions into the hub/dps exceptions or keep throwing them as-is?

Reply:
I think we should keep our exception list as narrow as possible. SwaggerException, HttpRequestException and CryptographicException should all fit into IotHub(Client/Service)Exception, since they arise out of communication attempts with IoT hub. As for the IndexOutOfRangeException, how deep in the stack do we throw this? Would it fit into our "parameter validation" exception?

Reply:
> As for the IndexOutOfRangeException, how deep in the stack do we throw this? Would it fit into our "parameter validation" exception?

The only 2 [places ](https://github.com/Azure/azure-iot-sdk-csharp/blob/previews/v2/provisioning/service/src/Models/Query.cs#L135
)of throwing `IndexOutOfRangeException` are both in the model "Query.cs". It may be fit into `InvalidOperationException` according to the use case here. Thoughts?

Reply:
For Query.cs, line 135 why not ArgumentNullException (null) and ArgumentException (whitespace)?

Reply:
For Query.cs line 153, I'd agree with InvalidOperationException.

Reply:
Yes, I agree.